libraries/litex-register-gen/Cargo.toml

```diff
@@ -0,0 +1,17 @@
+[package]
+name = "litex-register-gen"
```

Comment:
This is a long PR, maybe this could be split out separately?

Reply:
I agree that this should probably be reviewed separately. However swapping that out against the generated code won't exactly make the PR shorter, rather a bit longer. It'll be easier to review though.

Reply:
Done!